pkg/apis/discovery/types.go

```diff
@@ -0,0 +1,139 @@
+/*
+Copyright 2019 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package discovery
+
+import (
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	api "k8s.io/kubernetes/pkg/apis/core"
+)
+
+// +k8s:deepcopy-gen:interfaces=k8s.io/apimachinery/pkg/runtime.Object
+
+// EndpointSlice represents a subset of the endpoints that implement a service.
+// For a given service there may be multiple EndpointSlice objects, selected by
+// labels, which must be joined to produce the full set of endpoints.
+type EndpointSlice struct {
+	metav1.TypeMeta
+	// Standard object's metadata.
+	// +optional
+	metav1.ObjectMeta
+	// addressType specifies the type of address carried by this EndpointSlice.
+	// All addresses in this slice must be the same type.
+	// +optional
+	AddressType *AddressType
+	// endpoints is a list of unique endpoints in this slice. Each slice may
+	// include a maximum of 1000 endpoints.
+	// +listType=atomic
```

Comment:
Chances are, we will always replace a small set of endpoints (excepts in initial creation). Wouldn't it make make sense (perf wise) to patch merge this (this apply on all the lists in these new types).

Reply:
I am not sure how much impact would it have since the max number of endpoint per slides is 100 by default.